src/Nethermind/Nethermind.Runner/JsonRpc/Startup.cs

```diff
@@ -144,13 +147,14 @@ long SerializeTimeoutException(IJsonRpcService service, Stream resultStream)
                     await ctx.Response.WriteAsync("Nethermind JSON RPC");
                 }
 
-                if (ctx.Connection.LocalPort == jsonRpcConfig.Port && ctx.Request.Method == "POST")
+                JsonRpcUrl jsonRpcUrl = jsonRpcUrlCollection.FirstOrDefault(x => x.Port == ctx.Connection.LocalPort && x.RpcEndpoint.HasFlag(RpcEndpoint.Http));
```

Comment:
This is potentially slow, please create a Dictionary of http supported jsonRcpUrl's